tests/regression/src/lib.rs

```diff
@@ -55,20 +55,37 @@ mod tests {
             cg::cachegrind::start_instrumentation();
         }
     }
+
     /// Environment variable to determine whether to run under valgrind or solely test functionality.
     fn is_running_under_valgrind() -> bool {
         env::var("ENABLE_VALGRIND").is_ok()
     }
 
+    /// Function to get the test suffix from environment variables. If the var is not set, it defaults to curr.
+    fn get_test_suffix() -> String {
+        env::var("TEST_SUFFIX").unwrap_or_else(|_| "curr".to_string())
+    }
+
+    /// Function to determine if diff mode is enabled.
+    fn is_diff_mode() -> bool {
+        env::var("DIFF_MODE").is_ok()
+    }
+
     fn valgrind_test<F>(test_name: &str, test_body: F) -> Result<(), s2n_tls::error::Error>
     where
         F: FnOnce(&InstrumentationControl) -> Result<(), s2n_tls::error::Error>,
     {
+        let suffix = get_test_suffix();
         if !is_running_under_valgrind() {
-            let ctrl = InstrumentationControl;
-            test_body(&ctrl)
+            if is_diff_mode() {
+                run_diff_test(test_name);
+                Ok(())
+            } else {
+                let ctrl = InstrumentationControl;
+                test_body(&ctrl)
+            }
         } else {
-            run_valgrind_test(test_name);
+            run_valgrind_test(test_name, &suffix);
             Ok(())
```

Comment:
These changes are a great example of why it's a good idea to consolidate/abstract _how_ the tests are run from the harness definitions themselves. If we didn't have this abstraction, you'd have to go update every harness (and hopefully you consistently applied the change :))